portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -137,7 +141,7 @@ public void putInMap(List<BlogMessage> messages) {
           iterator = map.get(tag).iterator();
         }
         while(iterator.hasNext() && limit >= 0) {
-          values.addLast(iterator.next());
+          values.addLast((BlogMessage) iterator.next());
```

Comment:
Similarly, give iterator a template type (i..e, a BlogMessage) when you declare the "iterator" to avoid the conversion.